### PR TITLE
[CLI-439, WB-3445] Make logging output configurable

### PR DESCRIPTION
**Pull request by annirudh**
https://wandb.atlassian.net/browse/CLI-439
https://wandb.atlassian.net/browse/WB-3445

### CODE REVIEW

File: wandb/errors/term.py [hunk 1]
```diff
@@ -30,17 +63,10 @@ def termlog(string='', newline=True, repeat=True, prefix=True):
...
-    termlog(string=string, newline=True, **kwargs)
-
-
-def termerror(string, **kwargs):
-    string = '\n'.join(['{} {}'.format(ERROR_STRING, s)
-                        for s in string.split('\n')])
-    termlog(string=string, newline=True, **kwargs)
+    if silent:
+        util.mkdir_exists_ok(os.path.dirname(util.get_log_file_path()))
+        with open(util.get_log_file_path(), 'w') as log:
```
Comment (raubitsj) [inline wandb/errors/term.py:68]: I think `settings.log_user` is what you want here.  maybe capture that with termsetup()

File: wandb/errors/term.py [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
 
+_silent = False
+_show_info = True
+_show_warnings = True
+_show_errors = True
+
+
+def termsetup(settings):
+    global _silent, _show_info, _show_warnings, _show_errors
+    _silent = not settings._show_console
```
Comment (raubitsj) [inline wandb/errors/term.py:22]: show_console is confusing, maybe we should just have this be "silent" in settings also?

**Review by raubitsj — CHANGES_REQUESTED** [review]
looks great, a few small changes.

File: wandb/errors/term.py (hunk 1 above)
Comment (annirudh) [reply wandb/errors/term.py:68]: Won't that change the logs from going to `debug-internal.log` to `debug.log`? Just confirming that's what we want.

I copied this code as is from the original PR that introduced `WANDB_SILENT`: https://github.com/wandb/client/commit/245d53bfcb0dddd9036f49eb6a5834e02b77415b

Comment (annirudh) [reply wandb/errors/term.py:68]: I see that PR is super old. Will fix this.

Comment (raubitsj) [reply wandb/errors/term.py:68]: yes, debug.log is where logging from the user process goes,  which i think is what we want.

**Review by raubitsj — APPROVED** [review]

**Pull request merged**